linkml_runtime/dumpers/rdf_dumper.py

```diff
@@ -77,15 +78,19 @@ def dump(self, element: YAMLRoot, to_file: str, contexts: CONTEXTS_PARAM_TYPE =
             * A list containing elements of any type named above
         :param fmt: RDF format
         """
+        if isinstance(element, BaseModel):
+            element = element.dict()
         super().dump(element, to_file, contexts=contexts, fmt=fmt)
 
-    def dumps(self, element: YAMLRoot, contexts: CONTEXTS_PARAM_TYPE = None, fmt: Optional[str] = 'turtle') -> str:
+    def dumps(self, element: Union[BaseModel, YAMLRoot], contexts: CONTEXTS_PARAM_TYPE = None, fmt: Optional[str] = 'turtle') -> str:
         """
         Convert element into an RDF graph guided by the context(s) in contexts
         :param element: element to represent in RDF
         :param contexts: JSON-LD context(s) in the form of a file or URL, a json string or a json obj
         :param fmt: rdf format
         :return: rdflib Graph containing element
         """
+        if isinstance(element, BaseModel):
+            element = element.dict()
```

Comment:
This is the second part of the main theme of this PR, use pydantic's `.dict()` method on anything that is an instance of `BaseModel` and from there the code can carry on doing whatever it was doing and not need to care whether the instance was a BaseModel or a YAMLRoot